web/src/app/lists/ControlledPaginatedList.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { ReactElement, useState } from 'react'
+import {
+  Checkbox,
+  Grid,
+  Icon,
+  IconButton,
+  makeStyles,
+  Tooltip,
+} from '@material-ui/core'
+import {
+  PaginatedList,
+  PaginatedListItemProps,
+  PaginatedListProps,
+} from './PaginatedList'
+import { useSelector } from 'react-redux'
+import { urlKeySelector } from '../selectors/url'
+import classnames from 'classnames'
+import OtherActions from '../util/OtherActions'
+import { ArrowDropDown } from '@material-ui/icons'
+import Search from '../util/Search'
+
+const useStyles = makeStyles({
+  actionsContainer: {
+    alignItems: 'center',
+    display: 'flex',
+    marginRight: 'auto',
+    paddingLeft: '1em', // align with listItem icons
+    width: 'fit-content',
+  },
+  checkbox: {
+    marginTop: 4,
+    marginBottom: 4,
+  },
+  controlsContainer: {
+    alignItems: 'center',
+    display: 'flex',
+  },
+  hover: {
+    '&:hover': {
+      cursor: 'pointer',
+    },
+  },
+  popper: {
+    opacity: 1,
+  },
+  search: {
+    paddingLeft: '0.5em',
+  },
+})
+
+export interface ControlledPaginatedListProps extends PaginatedListProps {
+  checkboxActions?: ControlledPaginatedListAction[]
+  filter?: ReactElement
+
+  // if set, the search string param is ignored
+  noSearch?: boolean
+
+  // filters additional to search, set in the search text field
+  searchAdornment?: ReactElement
+
+  items: CheckboxItemsProps[] | PaginatedListItemProps[]
+}
+
+export interface ControlledPaginatedListAction {
+  // icon for the action (e.g. X for close)
+  icon: ReactElement
+
+  // label to display (e.g. "Close alerts")
+  label: string
+
+  // Callback that will be passed a list of selected items
+  onClick: (selectedIDs: (string | number)[]) => void
+}
+
+// used if checkBoxActions is set to true
+export interface CheckboxItemsProps extends PaginatedListItemProps {
+  type: 'checkbox'
```

Comment:
I'm actually using this as a user-defined type guard for Typescript checks between unions: https://github.com/target/goalert/pull/492/files/b8daef354923252ff14eeb8aa625e4ef7b9f9958#diff-8c55e9a99439ba82a0691ed298bfc0e8R111

It's not provided by the parent, so we are setting the type on this interface and if the type is present we know it's a checkbox. Otherwise it was fussing about how `id` and `selectable` may be undefined as `items` is a union type between `CheckboxItemsProps[]`  and `PaginatedListItemProps[]`

Reply:
Pushed up a fix